numpy/core/src/multiarray/alloc.c

```diff
@@ -25,10 +25,14 @@
 
 #include <assert.h>
 
-#ifdef HAVE_SYS_MMAN_H
+#ifdef NPY_OS_LINUX
 #include <sys/mman.h>
-#if defined MADV_HUGEPAGE && defined HAVE_MADVISE
-#define HAVE_MADV_HUGEPAGE
+#ifndef MADV_HUGEPAGE
+/*
+ * Use code 14 (MADV_HUGEPAGE) if it isn't defined. This gives a chance of
+ * enabling huge pages even if built with linux kernel < 2.6.38
+ */
```

Comment:
Just to clarify, this means that even if compiled on a machine running an old kernel, it will enable huge pages when run on a newer kernel?

Reply:
That's correct. An attempt to enable it will be made, and if it errors out, hugepages is simply ignored.